### ISSUE TITLE
Update prepare-packages script to remove dev dependencies, binary artifacts

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#61753]
We should only include the minimum set of packages needed for flutter run on the template. We should be able to exclude tool deps by uploading the snapshot, as well as the non-univeral binaries.

Blocked by https://github.com/flutter/flutter/issues/61751

**Comment by christopherfujino** [comment]
Will the strategy for this be to explicitly set which assets to clean/delete? or to make `flutter update-packages` more configurable as to what it installs? If we do the former, we may also need a benchmark to catch binary size regressions (I feel like I've seen an issue for that, but a cursory github search turned up negative).

**Comment by jonahwilliams** [comment]
We should opt-in/allowlist which artifacts are in included in the cache. The integration test will validate this is sufficient to build.

**Comment by christopherfujino** [comment]
> We should opt-in/allowlist which artifacts are in included in the cache. The integration test will validate this is sufficient to build.

SGTM

**Comment by christopherfujino** [comment]
although, that will mean we won't catch it until release time, and if there's a failure we'll need to hotfix the framework.

**Comment by christopherfujino** [comment]
i guess we could add an experimental "don't upload!" version of hte prepare_package job to master post-submit

**Comment by jonahwilliams** [comment]
> although, that will mean we won't catch it until release time, and if there's a failure we'll need to hotfix the framework.

No way! we can produce the zip file with any version - that will be part of the integrationtest.

**Comment by christopherfujino** [comment]
An easy win here would be to add a new flag to `flutter update-packages --public` that only runs `pub get` in directories under `//flutter/packages/`. This will remove ~150mb from the pub-cache just for flutter_gallery_assets and googleapis, which are only required from `//flutter/dev`.

**Comment by pcsosinski** [comment]
is this going to land in 1.26? if not, based on it's path through various milestones I'd argue it's not a P3

**Comment by jonahwilliams** [comment]
No, I don't think we've spent any time of this. We could revisit during OKR planning whether we have time to do this.

**Comment by itsjustkevin** [comment]
Thanks for the clarity on this @jonahwilliams! I will keep the issue open and triage it in our next meeting.
